src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -177,7 +177,19 @@ class ProtobufPlugin implements Plugin<Project> {
      */
     private void addSourceSetExtensions() {
       getSourceSets().all {  sourceSet ->
-        sourceSet.extensions.create('proto', ProtobufSourceDirectorySet, sourceSet.name, fileResolver)
+        String name = sourceSet.name
+        if (Utils.compareGradleVersion(project, "5.0") < 0) {
+          // TODO(zhangkun83): remove dependency on Gradle internal APIs once we drop support for < 5.0
+          sourceSet.extensions.create('proto', ProtobufSourceDirectorySet, sourceSet.name, fileResolver)
```

Comment:
Why do we need to use ProtobufSourceDirectorySet instead of DefaultSourceDirectorySet? It seems like using DefaultSourceDirectorySet here would make the else more clear. Backward compatibility?

Reply:
If we used DefaultSourceDirectorySet here, we would have to `configure()` it after it's created. Unfortunately, the only `configure()` available on Gradle 3.0 is
```
<T> void configure​(Class<T> type,
                   Action<? super T> action)
```
That means a custom type would still be necessary.

Reply:
Wait ... I don't actually need `configure()` to configure it, since I already have the reference to the SDS.